packages/hiro-graph-client/src/client.js

```diff
@@ -566,7 +566,7 @@ export default class Client {
     /**
      *  Write timeseries values (only ogit/Timeseries vertices)
      *
-     *  values are { timeseriesId: string, value: {timestamp: number; value: PlainObject | Array<PlainObject>}}
+     *  values are { timestamp: millisecond unix, value: string value }
```

Comment:
Well, there is stringifying inside of the client and I am curious can we have problems here if we stringify twice?
Coz this assumes one will call it as `client("id", {timestamp: 00000000, value: JSON.stringify({...})})` and then it will stringify again.. And then do we have to parse twice to get the initial json?

Reply:
Interesting! Previously only **kat-worker** passed not stringified object as value to writets function ->  [link](https://github.com/arago/hiro-desktop-utils/blob/master/packages/hiro-kat-worker/src/handlers/ki.ts#L138) 

But in **hiro-populate**, when timeseries values were generated, values with already stringified data were used, and there were no problems with these timeseries values in IV 🤔 (e.g. [here](https://github.com/arago/hiro-desktop-utils/blob/master/packages/hiro-populate-cli/src/handlers/common/preprocessor/nonExecuted.ts#L16)).

So, I think stringified type for timeseries items value is more correct, and it will be similar to the model from swagger.

Reply:
Hiro populate is not something we use in reality
And probably we are not really using data it generates as json objects

Reply:
Ok, I understand.
Just a brief history: there was no type for 'writets' function before, and I added it in pull https://github.com/arago/hiro-graph-js/pull/182 based on kat-worker code (value: object). Also, there was a [comment in client ](https://github.com/arago/hiro-graph-js/blob/425ffdddd094d694d2c2df7a9570d9e1b4d23df2/packages/hiro-graph-client/src/client.js#L569) that value is a string, but it did not work for kat-worker, so I added PlainObject to the interface.
And now I am confused a little bit about what was right: kat-worker or comment in client.js.

@vteremasov how do you this, does it work if the type will contain **any**, and then it will be stringified inside the client

```
writets: (
    timeseriesId: string,
    value:  { timestamp: number; value: any } | Array<{ timestamp: number; value: any }>
) => Promise<TimeseriesResponse[]>
```

Reply:
Any is bad practice and not allowed by our code styles. Also, I don't really like this ugly union type-notation. 

Can we try to make it using functions overloading? https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads

```ts
writets: (timeseriesId: string, value: PoJo) => Promise<TimeseriesResponse[]>;
writets: (timeseriesId: string, value: PoJo[]) => Promise<TimeseriesResponse[]>;
writets: (timeseriesId: string, value: string) => Promise<TimeseriesResponse[]>;
writets: (timeseriesId: string, value: string[]) => Promise<TimeseriesResponse[]>;
writets: (timeseriesId: string, value: number) => Promise<TimeseriesResponse[]>;
writets: (timeseriesId: string, value: number[]) => Promise<TimeseriesResponse[]>;
```

Reply:
thanks a lot for the suggestion, I will fix the interface and user overloading for this  function

Reply:
Overloading did not work, because in index.d.ts I've got an error `TS2300: Duplicate identifier 'writets'.` Also read a little TS documentation and they say to use union types whenever possible [(Do's and Don'ts)](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#use-union-types)
So I have created a type for TimeseriesValue where unions are used